phalcon/assets/collection.zep

```diff
@@ -49,7 +49,7 @@ class Collection implements \Countable, \Iterator
 
 	protected _attributes = [] { get };
 
-	protected _join = true { get };
+	protected _join = false { get };
```

Comment:
True, but fixing PR #12370 does that anyway. The changes result in joining of assets in those cases where no filters are applied, which it previously didn't. It had impact the result of 6(?) unit testcases. So having join() behave logically and how it is described in the documentation breaks Backward Compatibility anyhow. 

Which use case is used most often? Both filtering and joining? Then we should leave the default value true. On the other hand, not changing it will result in joined assets where developers previously have decided(?) not to join those assets. This made me decide to change the default value.

Alternatively one could introduce a new solution, like Phalcon\Asstes\CachedManager that deals with this issue and the changes requested in PR #1756. Then depreciate Phalcon\Assets\Manager or document it properly as it functions currently.